api/vpc_peering.go

```diff
@@ -82,26 +83,41 @@ func (api *API) ReadVpcPeeringRequest(instanceID int, peeringID string) (map[str
 	return data, nil
 }
 
-func (api *API) AcceptVpcPeering(instanceID int, peeringID string) (map[string]interface{}, error) {
-	_, err := api.waitForPeeringStatus(instanceID, peeringID)
-
+func (api *API) retryAcceptVpcPeering(instanceID int, peeringID string, attempt, sleep, timeout int) (map[string]interface{}, error) {
 	data := make(map[string]interface{})
 	failed := make(map[string]interface{})
-	log.Printf("[DEBUG] go-api::vpc_peering::accept instance id: %v, peering id: %v", instanceID, peeringID)
 	path := fmt.Sprintf("/api/instances/%v/vpc-peering/request/%v", instanceID, peeringID)
 	response, err := api.sling.New().Put(path).Receive(&data, &failed)
-	log.Printf("[DEBUG] go-api::vpc_peering::accept data: %v", data)
 
 	if err != nil {
 		return nil, err
 	}
-	if response.StatusCode != 200 {
+	if attempt*sleep >= timeout {
+		return nil, fmt.Errorf("AcceptVpcPeering failed, reach timeout")
+	} else if response.StatusCode == 400 {
+		errorCode := failed["error_code"].(float64)
+		if errorCode == 40001 {
+			log.Printf("[DEBUG] go-api::vpc_peering::accept firewall not finished configuring will retry "+
+				"accept VPC peering, attempt: %d, until timeout: %d", attempt, (timeout - (attempt * sleep)))
+			attempt++
+			time.Sleep(time.Duration(sleep) * time.Second)
+			return api.retryAcceptVpcPeering(instanceID, peeringID, attempt, sleep, timeout)
+		}
```

Comment:
I think we need an `else` here (response status was 400 but `error_code` was not `40001`)? 

with

    return nil, fmt.Errorf("AcceptVpcPeering failed, status: %v, message: %s", response.StatusCode, failed)

Reply:
Done. Linting suggested no need for else case, pure return instead.